modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -90,6 +90,16 @@ class StateChannel[F[_]: Async: KryoSerializer: SecurityProvider: Random](
     .filter(identity)
     .as(OwnRoundTrigger)
 
+  private val peerDiscovery: Stream[F, Unit] = Stream
+    .awakeEvery(10.seconds)
```

Comment:
I would go with less frequent check - not that much should change in 10s I think. We could also tie it to snapshot pulling (push snapshots to two separate pipelines) and check facilitators back then - maybe in case there are new facilitators that we don't know we should run the discovery and not pull them if no additional facilitators could be detected.

Reply:
There can be many joins or drops in a single 10s, especially when nodes are joining at the beginning after restart. It should not be strictly coupled with polling the global snapshot because changes can be in between as well, at least with the current algorithm. We agreed that the periodic check would be sufficient for now

Reply:
10s though is too frequent in my opinion anyway